### PR TITLE
Support both number and bignumber types when encoding numbers, add missing int/uint evm codecs

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: evm/evm-codec/src/codecs/primitives.ts [hunk 1]
```diff
@@ -14,119 +17,119 @@ export const bool: Codec<boolean> = {
   isDynamic: false,
 }
 
-export const uint8: Codec<number> = {
-  encode(sink: Sink, val: number) {
-    sink.u8(val)
+export const uint8: Codec<Numberish, number> = {
+  encode(sink: Sink, val: Numberish) {
+    sink.u8(safeToNumber(val, 0, 255))
```
Comment (belopash) [inline evm/evm-codec/src/codecs/primitives.ts:22]: @eldargab can we just check here if it is safeInteger and sink (view) will de the rest?

Comment (belopash) [reply evm/evm-codec/src/codecs/primitives.ts:22]: just to reduce overhead

Comment (eldargab) [reply evm/evm-codec/src/codecs/primitives.ts:22]: Yes, `DataView` set methods already check input boundaries.

Comment (belopash) [reply evm/evm-codec/src/codecs/primitives.ts]: As turned out it doesn't

**Pull request merged**